ETL/Python ETL/MatchHistoryETL.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+# In case iPython does not find our personalized modules and we want to import them manually
+# import sys
+# sys.path.append('my/path/to/module/folder')
+# import module_of_interest
+
+# We can also make sure what's the main directory iPhython consider for running
+# import os
+# os.getcwd()
+
+import constants
+import requests
+import pandas as pd
+import numpy as np
+import psycopg2
+import csv
+import datetime
+from psycopg2 import sql
+
+pd.set_option('display.max_rows', 100)
+pd.set_option('display.max_columns', 50)
+
+
+# In[2]:
+
+
+def calculate_matchweek(conn):
+    """
+        Finds out last registered matchweek for current season in Database. 
+        Returns last registerered matchweek plus one. If there's any problem 
+        with databse or season has not even started, then returns default matchweek "1"
+    """
+    try:
+        cursor = conn.cursor()
+        matchweek_query = f"SELECT MAX(\"MatchWeek\") FROM public.match_history WHERE \"Season\" = '{constants.CURRENT_SEASON_TAG}'"
+        cursor.execute(matchweek_query)
+        matchweek_result = cursor.fetchone()
+        if matchweek_result[0] is None:
+            return constants.DEFAULT_MATCHWEEK
+        else:
+            # Next matchweek
+            return matchweek_result[0] + 1
+    except psycopg2.Error as e:
+        print ("An error ocurred in database")
+        print (e)
+        return constants.DEFAULT_MATCHWEEK
+    finally:
+        cursor.close()
+
+
+# In[3]:
+
+
+def upsert_query(to_table, reference_column, dataframe):
```

Comment:
interesting to see you've written a function here for creating queries. Why not have a directory for SQL queries and call them using filepath

Reply:
Ok that's interesting approach. In this case I need to generate a dynamic query using python composable objects. But yeah, I can create a module  with template queries and functions like you mention to avoid writing functions in same script. Thanks for that observation